src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1383,6 +1383,15 @@ public static bool IsAtStartOfPattern(this SyntaxTree syntaxTree, SyntaxToken le
                 return leftToken.Parent is BinaryPatternSyntax;
             }
 
+            // e is SomeEnum.SomeEnumValue and $$
+            // e is SomeEnum.SomeEnumValue or $$
+            // 'and' & 'or' are identifier here because of lack of context
+            if (leftToken.IsKind(SyntaxKind.IdentifierToken) &&
+                (leftToken.IsKindOrHasMatchingText(SyntaxKind.AndKeyword) || leftToken.IsKindOrHasMatchingText(SyntaxKind.OrKeyword)))
```

Comment:
What happens if this became just this:

```suggestion
            if (leftToken.IsKindOrHasMatchingText(SyntaxKind.AndKeyword) || leftToken.IsKindOrHasMatchingText(SyntaxKind.OrKeyword))
```

Anything will break? If yes, consider making sure there is a test that will break if this change is made.